projects/js-packages/connection/components/disconnect-dialog/index.jsx

```diff
@@ -200,6 +200,11 @@ const DisconnectDialog = props => {
 		e => {
 			e && e.preventDefault();
 
+			jetpackAnalytics.tracks.recordEvent(
+				'jetpack_disconnect_dialog_click_disconnect',
+				defaultTracksArgs
+			);
+
```

Comment:
What would you think about having the disconnect button using a local event callback method inside the `<StepDisconnect />` component? That could allow a click on the disconnect button to call both `onDisconnect()` as well as `trackModalClick()`.
This could keep the analytics code in the `trackModalClick` function and also allow for clicking on the Disconnect button to be added to the test suite.

Reply:
Good idea - done.